db/crud.go

```diff
@@ -111,6 +111,11 @@ func (db *DatabaseContext) GetDocWithXattr(key string, unmarshalLevel DocumentUn
 	if unmarshalErr != nil {
 		return nil, nil, unmarshalErr
 	}
+
+	if len(rawBucketDoc.UserXattr) > 0 {
+		doc.rawUserXattr = rawBucketDoc.UserXattr
```

Comment:
Would it make more sense to do this inside unmarshalDocumentWIthXattr?  There seem to be other callers of unmarshalDocumentWithXattr (updateAndReturnDoc, UpdateAllDocChannels at least) that would want to do this, and it would be ignored if a value wasn't provided for the user xattr.

Reply:
Yeah thats a fair point. Will push it down to that.